app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
@liyakun Multiple files is probably not necessary at this point. Dropzone can limit the number of files to 1 (see http://stackoverflow.com/questions/18048825/how-to-limit-the-number-of-dropzone-js-files-uploaded/18156897#18156897).

@73 This feature is not ready to be merged, but it will be when it has been properly implemented.